tests/testthat/test-average_wait.R

```diff
@@ -0,0 +1,56 @@
+# # Anticipated test from the error handling that Matt Dray is drafting
+# test_that("function catches null values and reports error", {
+#   em <- "target_wait(): no error message when function is run with no inputs."
+#   expect_error(target_wait(), em)
+# })
+#
+# # Anticipated test from the error handling that Matt Dray is drafting
+# test_that("function catches mismatched input lengths", {
+#   em <- "target_wait(): no error message when functions inputs are of different length."
+#   expect_error(target_wait(c(22,25,26), c(4, 3)), em)
+# })
+#
+# # Anticipated test from the error handling that Matt Dray is drafting
+# test_that("it returns an error if either input aren't numeric", {
+#   in1 <- Sys.Date()
+#   in2 <- 4
+#
+#   em <- "target_wait(): all inputs must be numeric."
+#   expect_error(target_wait(in1, in2), em)
+# })
+
+test_that("it returns an expected result with fixed single values, against arithmetic", {
+  target_wait <-
+    em <- "average_wait(): aritmetic error with single value inputs."
```

Comment:
Again, might be misunderstanding, but what purpose do these `em` objects serve? I see they're in the format of the `e`rror `m`essages that the tests emit if the test fails.

Reply:
Also, whoops: 'aritmetic'.

Reply:
I'll confess, I pinched it from Tom Jemmett's tests in NHSRplotthedots.  I don't know if it's go practice or not.  Will correct my spelling.

Reply:
I've dropped Tom a line; not a dealbreaker!

Reply:
I'm only seeing part of the line on my phone, but if you are testing code that emits an error you may want to check that it is the error type you are expecting and not some other error; passing in the expected error message to expect_error can do this for you.

Whether you inline that into the expect_error or have the expectation as a variable is just style preference

Reply:
Interesting, thank you!